src/kernel/chainparams.cpp

```diff
@@ -508,6 +511,7 @@ class CRegTestParams : public CChainParams
         base58Prefixes[EXT_SECRET_KEY] = {0x04, 0x35, 0x83, 0x94};
 
         bech32_hrp = "bcrt";
+        silent_payment_hrp = "sprt";
```

Comment:
This issue has some helpful context: https://github.com/bitcoin/bitcoin/issues/12314#issuecomment-582600899

TLDR; at the time regtest was created, it was assumed that using a separate HRP wouldn't cost anything.

> If each of the four had a different one, I’d get it, but why does regtest have a different one than the other two testing networks?

This actually makes sense to me. Testnet and Signet are "global" networks, so everyone has to agree on the address format. Regtest is always used in a private context, and is specific to Bitcoin Core. Having a separate HRP helps differentiate the local vs networked usage IMO.

> Unless I’m forgetting about some good reason for that to be there, I might recommend that you just use tsp for all three

If we decided to make the HRP for regtest throughout Bitcoin Core to be the same as signet and testnet, then I'm fine with this. What I don't want is for silent payment addresses to be a special case in the functional test framework.

Reply:
As long as testnet and signet are the same, and different from mainnet, I'm happy. When testing with e.g. hardware wallets it lets you use the testnet app / firmware instead of signet-specific stuff.

Reply:
@josibake: It’s rather that bech32(m) addresses are the odd ones out. Before native segwit outputs all addresses across testnet and regtest were the same. After rereading #12314, I would still recommend that you use the same hrp for all test-networks.

Reply:
In the interest of keeping this PR well scoped, I think we should go for consistency. Since silent payment addresses are bech32m encoded, we should follow the convention _currently_ used for bech32(m), and if we do decided to change the convention for bech32(m) hrp's in the future, then it should change for all of them (including silent payments)